src/shogun/features/StringFeatures.cpp

```diff
@@ -160,10 +160,9 @@ template<class ST> SGVector<ST> StringFeatures<ST>::get_feature_vector(int32_t n
 	int32_t l;
 	bool free_vec;
 	ST* vec=get_feature_vector(num, l, free_vec);
-	ST* dst=SG_MALLOC(ST, l);
-	sg_memcpy(dst, vec, l*sizeof(ST));
```

Comment:
you should not remove this as without this the content of `dst` vector is nothing (or all 0s to be precise)

Reply:
So do I leave it with the Malloc there

Reply:
you can also do `sg_memcpy(dst.vector, vec, l*sizeof(ST))`;

Reply:
@Khalifa1997 you should use the SGVector constructor with only the size parameter: `SGVector<ST>dst(l);` and then copy the data over: `sg_memcpy(dst.vector, vec, l*sizeof(ST));`
Right now the data in `vec` that you pass to the sgvector constructor gets deleted below (`free_feature_vector(vec, num, free_vec);`), so you need that copy. And then the sgvector will also try deallocate the memory that was deleted by `free_feature_vector`...

Reply:
Got it! thanks